xarray/core/common.py

```diff
@@ -381,11 +382,11 @@ class DataWithCoords(AttrAccessMixin):
     __slots__ = ("_close",)
 
     def squeeze(
-        self: T_DataWithCoords,
+        self,
         dim: Hashable | Iterable[Hashable] | None = None,
         drop: bool = False,
         axis: int | Iterable[int] | None = None,
-    ) -> T_DataWithCoords:
+    ) -> Self:
```

Comment:
Something I've been wondering about with Self usage: Is a copied version of self ok? 

`squeeze` uses `isel` which in the end creates a new copy with `types(self)(dims=...,  data=...)`

Reply:
IIUC, it means that it's the _type_ of `self`, not that it's literally the object. So a copy would be completely fine, given it retains its type. (or tell me if I'm misunderstanding)

Reply:
Nice, then there's a lot of places that could use Self!